src/huggingface_hub/hub_mixin.py

```diff
@@ -19,30 +19,154 @@
+            >>> model_path: str = file_manager.get_filepath("pytorch_model.bin")
+            >>> config: dict = file_manager.load_json_file('config.json')
+        """
+
+        self.model_id = pretrained_model_name_or_path
+        self.cache_dir = cache_dir
+        self.force_download = force_download
+        self.proxies = proxies
+        self.resume_download = resume_download
+        self.local_files_only = local_files_only
+        self.use_auth_token = use_auth_token
+
+        self.revision = None
+        if len(pretrained_model_name_or_path.split("@")) == 2:
+            self.model_id, self.revision = pretrained_model_name_or_path.split("@")
 
-            >>> class MyModel(nn.Module, ModelHubMixin):
-            ...    def __init__(self, **kwargs):
-            ...        super().__init__()
-            ...        self.config = kwargs.pop("config", None)
-            ...        self.layer = ...
-            ...    def forward(self, ...)
-            ...        return ...
+        self.model_dir = Path(pretrained_model_name_or_path)
 
-            >>> model = MyModel()
-            >>> model.save_pretrained("mymodel", push_to_hub=False) # Saving model weights in the directory
-            >>> model.push_to_hub("mymodel", "model-1") # Pushing model-weights to hf-hub
+    def get_filepath(self, filename: str):
+        """
+        Get the filepath to a given file found in model_dir or cache_dir. If not initially found in either, attempt to
+        download it from HuggingFace hub and save it to cache_dir.
+
+        Parameters:
+            filename (:obj:`str`):
+                Name of file expected to be in model_dir, cache_dir, or on remote huggingface hub repo.
+        Returns:
+            The local path to this file. If not found in model_dir/cache_dir, it will be downloaded from the hub
+            and saved to cache_dir. If it does not exist, we return None.
+        """
+        potential_local_filepath = self.model_dir / filename
+        if self.model_dir.is_dir() and potential_local_filepath.exists():
+            filepath = potential_local_filepath.as_posix()
+        else:
+            try:
+                file_url = hf_hub_url(
+                    self.model_id, filename=filename, revision=self.revision
+                )
+                filepath = cached_download(
+                    file_url,
+                    cache_dir=self.cache_dir,
+                    force_download=self.force_download,
+                    proxies=self.proxies,
+                    resume_download=self.resume_download,
+                    local_files_only=self.local_files_only,
+                    use_auth_token=self.use_auth_token,
+                )
+            except requests.exceptions.RequestException:
+                logger.warning(f"{filename} NOT FOUND in HuggingFace Hub")
+
+                # TODO - decide if we should error here instead of returning None
```

Comment:
I'd really like to just error out here if you can't find it in `save_directory`, `cache_dir` or remotely on the hub. Is there any reason why not that I'm missing?